shutdown.go

```diff
@@ -74,10 +105,61 @@ func (app *App) broadcastSignal(signal os.Signal) error {
 	}
 
 	if unsent != 0 {
-		return fmt.Errorf("failed to send %v signal to %v out of %v channels",
-			signal, unsent, len(app.dones),
-		)
+		return ErrOnUnsentSignal{
+			Signal:   signal,
+			Unsent:   unsent,
+			Channels: len(app.dones),
+		}
 	}
 
 	return nil
 }
+
+func (app *App) broadcastWaitSignal(signal os.Signal, code int) error {
+	app.waitsMu.Lock()
+	defer app.waitsMu.Unlock()
+
+	app.shutdownSignal = &ShutdownSignal{
+		Signal:   signal,
+		ExitCode: code,
+	}
+
+	var unsent int
+	for _, wait := range app.waits {
+		select {
+		case wait <- *app.shutdownSignal:
+		default:
+			// shutdown called when wait channel has already received a
+			// termination signal that has not been cleared
+			unsent++
+		}
+	}
+
+	if unsent != 0 {
+		return ErrOnUnsentSignal{
+			Signal:   signal,
+			Unsent:   unsent,
+			Code:     code,
+			Channels: len(app.waits),
+		}
+	}
+
+	return nil
+}
+
+// ErrOnUnsentSignal ... TBD
+type ErrOnUnsentSignal struct {
+	Signal   os.Signal
+	Unsent   int
+	Code     int
+	Channels int
+}
+
+func (err ErrOnUnsentSignal) Error() string {
+	return fmt.Sprintf(
+		"failed to send %v signal to %v out of %v channels",
```

Comment:
I think the error message can be more concise.
This error message was written when we hadn't internalized the error message guidance of omitting "failed to x" and "error doing x" from messages. (This is now part of our style guide.)

How about something like:

    send %v signal: %v/%v channels are blocked